File: patterns/java/src/refactoring_guru/patterns/iterator/example/social_networks/SocialNetwork.java
package refactoring_guru.patterns.iterator.example.social_networks;

import refactoring_guru.patterns.iterator.example.iterators.ProfileIterator;

public interface SocialNetwork {
    public ProfileIterator getFriendsIterator(String profileEmail);

    public ProfileIterator getCoworkersIterator(String profileEmail);
}

File: patterns/cs/src/RefactoringGuru/Iterator/Example/SocialNetworks/SocialNetwork.cs
using System;
using RefactoringGuru.Iterator.Example.Iterators;

namespace RefactoringGuru.Iterator.Example.SocialNetworks
{
    interface SocialNetwork
    {
        ProfileIterator GetFriendsIterator(String profileEmail);

        ProfileIterator GetCoworkersIterator(String profileEmail);
    }
}